### PR TITLE
Issue in wallet field types

### PR DESCRIPTION
**Pull request by remybar**
At the end of the chapter 8.1, 2 `Wallet` are instanciated with the following code:

```
    let w1 = Wallet { balance: true, address: 10 };
    let w2 = Wallet { balance: 32, address: 100 };
```

Then, a paragraph explains that we create 2 instances: one of `Wallet<bool, u128>` and the other of `Wallet<felt252, u8>`.

I think that it's not correct as the compiler will use `felt252` to store `10`, `32` and `100` values by default.

To be sure to have an instance of `Wallet<bool, u128>` and another of `Wallet<felt252, u8>`, we have to use `10_u128` and `100_u8`.

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/490)

### CODE REVIEW

File: listings/ch08-generic-types-and-traits/no_listing_14_compiling/src/lib.cairo [hunk 1]
```diff
@@ -22,8 +22,8 @@ impl WalletMixImpl<T1, +Drop<T1>, U1, +Drop<U1>> of WalletMixTrait<T1, U1> {
 
 // ANCHOR: main
 fn main() {
-    let w1 = Wallet { balance: true, address: 10 };
-    let w2 = Wallet { balance: 32, address: 100 };
+    let w1 = Wallet { balance: true, address: 10_u128 };
```
Comment (enitrat) [inline listings/ch08-generic-types-and-traits/no_listing_14_compiling/src/lib.cairo:25]:
```suggestion
    let w1: Wallet<bool, u128> = Wallet { balance: true, address: 10_u128 };
```

File: listings/ch08-generic-types-and-traits/no_listing_14_compiling/src/lib.cairo [hunk 2]
```diff
@@ -22,8 +22,8 @@ impl WalletMixImpl<T1, +Drop<T1>, U1, +Drop<U1>> of WalletMixTrait<T1, U1> {
...
+    let w2 = Wallet { balance: 32, address: 100_u8 };
```
Comment (enitrat) [inline listings/ch08-generic-types-and-traits/no_listing_14_compiling/src/lib.cairo:26]:
```suggestion
    let w2: Wallet<bool, u8> = Wallet { balance: 32, address: 100_u8 };
```

**Review by enitrat — COMMENTED** [review]
Great, nice catch. However I'd rather have explicit type annotation, for clarity, rather than relying on automatic type inference. It makes it more obvious to the reader as well.

Thank you for your thorough reviews!

**Comment by remybar** [comment]
> Great, nice catch. However I'd rather have explicit type annotation, for clarity, rather than relying on automatic type inference. It makes it more obvious to the reader as well.
> 
> Thank you for your thorough reviews!

Yes, I agree 👍

**Review by enitrat — APPROVED** [review]

**Pull request merged**